zookeeper-server/src/test/java/org/apache/zookeeper/test/BackupManagerTest.java

```diff
@@ -0,0 +1,601 @@
+    File dataDir = ClientBase.createTmpDir();
+    File backupTmpDir = ClientBase.createTmpDir();
+    File backupDir = ClientBase.createTmpDir();
+
+    BackupManager bm = new BackupManager(dataDir, dataDir, dataDir, backupTmpDir, 15,
+        new LocalBackupStorage(backupConfig));
+    bm.initialize();
+
+    bm.getLogBackup().run(1);
+    bm.getSnapBackup().run(1);
+
+    Assert.assertFalse("No files should have been backed up.", backupDir.list() == null);
+    Assert.assertFalse("No temporary files should have been created.",
+        backupTmpDir.list() == null);
+    BackupStatus bs = new BackupStatus(dataDir);
+    BackupPoint bp = bs.read();
+    Assert.assertEquals("backupStatus should be set to min for log",
+        bp.getLogZxid(), BackupUtil.INVALID_LOG_ZXID);
+    Assert.assertEquals("backupStatus should be set to min for snap",
+        bp.getSnapZxid(), BackupUtil.INVALID_SNAP_ZXID);
+  }
+
+  @Test
+  public void testSwitchToNewBackupNode() throws Exception {
+    createNodes(connection, 1000);
+    backupManager.getLogBackup().run(1);
+    backupManager.getSnapBackup().run(1);
+
+    BackupPoint bp = backupStatus.read();
+    String[] filesOrig = getBackupFiles(backupDir);
+    Assert.assertEquals("backup status and backupmanager must match log",
+        bp.getLogZxid(), backupManager.getBackedupLogZxid());
+    Assert.assertEquals("backup status and backupmanager must match snap",
+        bp.getSnapZxid(), backupManager.getBackedupSnapZxid());
+
+    File backupStatusFile = new File(backupStatusDir, BackupStatus.STATUS_FILENAME);
+    backupStatusFile.delete();
+
+    backupManager.initialize();
+    BackupPoint bp2 = backupStatus.read();
+    Assert.assertEquals(
+        "backup status and backupmanager must match log after switch to new backup node",
+        bp2.getLogZxid(),
+        backupManager.getBackedupLogZxid());
+    Assert.assertEquals(
+        "backup status and backupmanager must match snap after switch to new backup node",
+        bp2.getSnapZxid(),
+        backupManager.getBackedupSnapZxid());
+    Assert.assertEquals("backup point must match prior backup point -- snap",
+        bp.getLogZxid(), bp2.getLogZxid());
+    Assert.assertEquals("backup point must match prior backup point -- snap",
+        bp.getSnapZxid(), bp2.getSnapZxid());
+
+    backupManager.getLogBackup().run(1);
+    backupManager.getSnapBackup().run(1);
+
+    String[] filesPost = getBackupFiles(backupDir);
+    Assert.assertArrayEquals("files before and after switch to new backup node",
+        filesOrig, filesPost);
+
+    createNodes(connection, 1000);
+    backupManager.getLogBackup().run(1);
+    backupManager.getSnapBackup().run(1);
+
+    String[] filesPostCreates = getBackupFiles(backupDir);
+    List<Range<Long>> zxids = BackupUtil.getZxids(backupStorage, BackupFileType.TXNLOG);
+
+    Assert.assertTrue("must have more backups", filesPostCreates.length > filesPost.length);
```

Comment:
Could you please explain how do we make sure that we will have more backup files after the switch than before? I think the new transactions may be written in the same file, rather than a new log file. Probably should roll log every time?

Reply:
`SyncRequestProcessor.setSnapCount(100);` makes sure there the rollLog gets called at least once during setup. We are triggering `1000` operations with `createNodes(connection, 1000);`.